server/game/cards/06-WoE/EndlessHordes.js

```diff
@@ -0,0 +1,81 @@
+const Card = require('../../Card.js');
+
+class EndlessHordes extends Card {
+    // Play: For each enemy creature, make a token creature, ready it,
+    // and have it fight that enemy creature, ignoring taunt. Resolve
+    // these fights one at a time.
+    //
+    // Note the FAQ entry states that all the tokens are created first,
+    // then all readied, then fight targets are chosen one at a time and
+    // the fights are fully resolved.
+    //
+    // Unofficial ruling from blinkingline is that only enemy
+    // creatures that were present at the beginning of resolution can
+    // be targeted for later fighting.  If multiple creatures are
+    // killed during one fight, then the leftover tokens just end up
+    // ready (even if new creatures were created during earlier
+    // fights).
+    setupCardAbilities(ability) {
+        this.validTargets = [];
+        this.previousTokens = [];
+        this.newTokens = [];
+
+        this.play({
+            gameAction: ability.actions.makeTokenCreature((context) => {
+                this.validTargets = context.player.opponent.creaturesInPlay.slice();
+                this.previousTokens = context.player.creaturesInPlay.filter((card) =>
+                    card.isToken()
+                );
+                return {
+                    amount: this.validTargets.length
+                };
+            }),
+            then: {
+                gameAction: ability.actions.sequential([
+                    ability.actions.sequentialForEach((context) => {
+                        this.newTokens = context.player.creaturesInPlay
+                            .filter((card) => card.isToken())
+                            .filter((card) => !this.previousTokens.includes(card));
+
+                        return {
+                            forEach: this.newTokens,
+                            action: ability.actions.ready()
+                        };
+                    }),
+                    ability.actions.sequentialForEach({
+                        forEach: this.newTokens,
+                        gameAction: ability.actions.cardLastingEffect({
+                            effect: ability.effects.ignores('taunt')
```

Comment:
Granting this effect to each token is not working, and I can't figure out why.  Help please!

(There's probably another issue here, making it only last until the end of the card resolution, but I want to figure out how to get it to work at all first.)

Reply:
Ok, figured this out.  Had to add a new notion of canceling ignores, since there's no way to make effects expire once the card's play function finishes.

Reply:
I think you need to add a " ignoreTaunt " property to fight action

Reply:
Good suggestion, that's much nicer.  Done!